python3Packages.pymatting: fails to build with cudaSupport = true

### Nixpkgs version

`3cbe716e2346710d6e1f7c559363d14e11c32a43`

### Describe the bug

running following command:
```sh
nix build --impure --expr '
  (import (builtins.getFlake "github:nixos/nixpkgs/3cbe716e2346710d6e1f7c559363d14e11c32a43") {
    config = {
      cudaSupport = true;
      allowUnfree = true;
    };
  }).python3Packages.pymatting
'
```

fails with during test phase, with:

```console
error: Cannot build '/nix/store/99hkkypdszdbjcx8z0jhsmrzkpv6j7gv-python3.13-pymatting-1.1.13.drv'.
       Reason: builder failed with exit code 1.
       Output paths:
         /nix/store/1pmnl455svg4gidn6nh3gsrzglzf3cbb-python3.13-pymatting-1.1.13
         /nix/store/crb44v00j827vbyhyal95lwwzi7l8vs7-python3.13-pymatting-1.1.13-dist
       Last 25 log lines:
       >         }
       >     
       >         float a00 = a0 * a0 + a_sum;
       >         float a11 = a1 * a1 + a_sum;
       >         float a01 = a0 * a1;
       >     
       >         float inv_det = 1.0f / (a00 * a11 - a01 * a01);
       >     
       >         F[i * 3 + 0] = fmax(0.0f, fmin(1.0f, inv_det * (a11 * b00 - a01 * b10)));
       >         F[i * 3 + 1] = fmax(0.0f, fmin(1.0f, inv_det * (a11 * b01 - a01 * b11)));
       >         F[i * 3 + 2] = fmax(0.0f, fmin(1.0f, inv_det * (a11 * b02 - a01 * b12)));
       >         B[i * 3 + 0] = fmax(0.0f, fmin(1.0f, inv_det * (a00 * b10 - a01 * b00)));
       >         B[i * 3 + 1] = fmax(0.0f, fmin(1.0f, inv_det * (a00 * b11 - a01 * b01)));
       >         B[i * 3 + 2] = fmax(0.0f, fmin(1.0f, inv_det * (a00 * b12 - a01 * b02)));
       >     }
       >     """
       >     
       > >   platform = cl.get_platforms()[0]
       >                ^^^^^^^^^^^^^^^^^^
       > E   pyopencl._cl.LogicError: clGetPlatformIDs failed: PLATFORM_NOT_FOUND_KHR
       >
       > pymatting/foreground/estimate_foreground_ml_pyopencl.py:97: LogicError
       > =========================== short test summary info ============================
       > FAILED tests/test_foreground.py::test_foreground - pyopencl._cl.LogicError: clGetPlatformIDs failed: PLATFORM_NOT_FOUND_KHR
       > ================== 1 failed, 8 passed, 4 deselected in 38.73s ==================
       For full logs, run:
         nix log /nix/store/99hkkypdszdbjcx8z0jhsmrzkpv6j7gv-python3.13-pymatting-1.1.13.drv
```

### Steps to reproduce

as above

### Expected behaviour

it should not error

### Screenshots

_No response_

### Relevant log output

```console
...
python3.13-pymatting> Successfully built pymatting-1.1.13-py3-none-any.whl
python3.13-pymatting> Finished creating a wheel...
python3.13-pymatting> Finished executing pypaBuildPhase
python3.13-pymatting> Running phase: pythonRuntimeDepsCheckHook
python3.13-pymatting> Executing pythonRuntimeDepsCheck
python3.13-pymatting> Checking runtime dependencies for pymatting-1.1.13-py3-none-any.whl
python3.13-pymatting> Finished executing pythonRuntimeDepsCheck
python3.13-pymatting> Running phase: installPhase
python3.13-pymatting> Executing pypaInstallPhase
python3.13-pymatting> Successfully installed pymatting-1.1.13-py3-none-any.whl
python3.13-pymatting> Finished executing pypaInstallPhase
python3.13-pymatting> Running phase: pythonOutputDistPhase
python3.13-pymatting> Executing pythonOutputDistPhase
python3.13-pymatting> Finished executing pythonOutputDistPhase
python3.13-pymatting> Running phase: fixupPhase
python3.13-pymatting> shrinking RPATHs of ELF executables and libraries in /nix/store/1pmnl455svg4gidn6nh3gsrzglzf3cbb-python3.13-pymatting-1.1.13
python3.13-pymatting> checking for references to /build/ in /nix/store/1pmnl455svg4gidn6nh3gsrzglzf3cbb-python3.13-pymatting-1.1.13...
python3.13-pymatting> patching script interpreter paths in /nix/store/1pmnl455svg4gidn6nh3gsrzglzf3cbb-python3.13-pymatting-1.1.13
python3.13-pymatting> stripping (with command strip and flags -S -p) in  /nix/store/1pmnl455svg4gidn6nh3gsrzglzf3cbb-python3.13-pymatting-1.1.13/lib
python3.13-pymatting> shrinking RPATHs of ELF executables and libraries in /nix/store/crb44v00j827vbyhyal95lwwzi7l8vs7-python3.13-pymatting-1.1.13-dist
python3.13-pymatting> checking for references to /build/ in /nix/store/crb44v00j827vbyhyal95lwwzi7l8vs7-python3.13-pymatting-1.1.13-dist...
python3.13-pymatting> patching script interpreter paths in /nix/store/crb44v00j827vbyhyal95lwwzi7l8vs7-python3.13-pymatting-1.1.13-dist
python3.13-pymatting> Executing pythonRemoveTestsDir
python3.13-pymatting> Finished executing pythonRemoveTestsDir
python3.13-pymatting> Running phase: installCheckPhase
python3.13-pymatting> no Makefile or custom installCheckPhase, doing nothing
python3.13-pymatting> Running phase: pythonCatchConflictsPhase
python3.13-pymatting> Running phase: pythonRemoveBinBytecodePhase
python3.13-pymatting> Running phase: pythonImportsCheckPhase
python3.13-pymatting> Executing pythonImportsCheckPhase
python3.13-pymatting> Check whether the following modules can be imported: pymatting
python3.13-pymatting> Running phase: pytestCheckPhase
python3.13-pymatting> Executing pytestCheckPhase
python3.13-pymatting> pytest flags: -m pytest -k not\ \(test_alpha\)\ and\ not\ \(test_laplacians\)\ and\ not\ \(test_preconditioners\)\ and\ not\ \(test_lkm\)
python3.13-pymatting> ============================= test session starts ==============================
python3.13-pymatting> platform linux -- Python 3.13.7, pytest-8.4.1, pluggy-1.6.0
python3.13-pymatting> rootdir: /build/source
python3.13-pymatting> collected 13 items / 4 deselected / 9 selected                                 
python3.13-pymatting> 
python3.13-pymatting> tests/test_boxfilter.py .                                                [ 11%]
python3.13-pymatting> tests/test_cg.py .                                                       [ 22%]
python3.13-pymatting> tests/test_distance.py .                                                 [ 33%]
python3.13-pymatting> tests/test_foreground.py F                                               [ 44%]
python3.13-pymatting> tests/test_ichol.py .                                                    [ 55%]
python3.13-pymatting> tests/test_kdtree.py .                                                   [ 66%]
python3.13-pymatting> tests/test_remove_background_bicolor.py .                                [ 77%]
python3.13-pymatting> tests/test_simple_api.py .                                               [ 88%]
python3.13-pymatting> tests/test_util.py .                                                     [100%]
python3.13-pymatting> 
python3.13-pymatting> =================================== FAILURES ===================================
python3.13-pymatting> _______________________________ test_foreground ________________________________
python3.13-pymatting> 
python3.13-pymatting>     def test_foreground():
python3.13-pymatting>         try:
python3.13-pymatting>             from pymatting.foreground.estimate_foreground_ml_cupy import (
python3.13-pymatting>                 estimate_foreground_ml_cupy,
python3.13-pymatting>             )
python3.13-pymatting> >           from pymatting.foreground.estimate_foreground_ml_pyopencl import (
python3.13-pymatting>                 estimate_foreground_ml_pyopencl,
python3.13-pymatting>             )
python3.13-pymatting> 
python3.13-pymatting> tests/test_foreground.py:16:
python3.13-pymatting> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.13-pymatting> 
python3.13-pymatting>     import numpy as np
python3.13-pymatting>     import pyopencl as cl
python3.13-pymatting>     
python3.13-pymatting>     source = """
python3.13-pymatting>     __kernel void resize_nearest(
python3.13-pymatting>         __global float *dst,
python3.13-pymatting>         __global const float *src,
python3.13-pymatting>         int w_src, int h_src,
python3.13-pymatting>         int w_dst, int h_dst,
python3.13-pymatting>         int depth
python3.13-pymatting>     ){
python3.13-pymatting>         int x_dst = get_global_id(0);
python3.13-pymatting>         int y_dst = get_global_id(1);
python3.13-pymatting>     
python3.13-pymatting>         if (x_dst >= w_dst || y_dst >= h_dst) return;
python3.13-pymatting>     
python3.13-pymatting>         int x_src = min(x_dst * w_src / w_dst, w_src - 1);
python3.13-pymatting>         int y_src = min(y_dst * h_src / h_dst, h_src - 1);
python3.13-pymatting>     
python3.13-pymatting>         __global       float *ptr_dst = dst + (x_dst + y_dst * w_dst) * depth;
python3.13-pymatting>         __global const float *ptr_src = src + (x_src + y_src * w_src) * depth;
python3.13-pymatting>     
python3.13-pymatting>         for (int channel = 0; channel < depth; channel++){
python3.13-pymatting>             ptr_dst[channel] = ptr_src[channel];
python3.13-pymatting>         }
python3.13-pymatting>     }
python3.13-pymatting>     
python3.13-pymatting>     __kernel void ml_iteration(
python3.13-pymatting>         __global       float *F,
python3.13-pymatting>         __global       float *B,
python3.13-pymatting>         __global const float *F_prev,
python3.13-pymatting>         __global const float *B_prev,
python3.13-pymatting>         __global const float *image,
python3.13-pymatting>         __global const float *alpha,
python3.13-pymatting>         int w,
python3.13-pymatting>         int h,
python3.13-pymatting>         float regularization
python3.13-pymatting>     ){
python3.13-pymatting>         int x = get_global_id(0);
python3.13-pymatting>         int y = get_global_id(1);
python3.13-pymatting>     
python3.13-pymatting>         int i = x + y * w;
python3.13-pymatting>     
python3.13-pymatting>         if (x >= w || y >= h) return;
python3.13-pymatting>     
python3.13-pymatting>         float a0 = alpha[i];
python3.13-pymatting>         float a1 = 1.0f - a0;
python3.13-pymatting>     
python3.13-pymatting>         float b00 = a0 * image[i * 3 + 0];
python3.13-pymatting>         float b01 = a0 * image[i * 3 + 1];
python3.13-pymatting>         float b02 = a0 * image[i * 3 + 2];
python3.13-pymatting>     
python3.13-pymatting>         float b10 = a1 * image[i * 3 + 0];
python3.13-pymatting>         float b11 = a1 * image[i * 3 + 1];
python3.13-pymatting>         float b12 = a1 * image[i * 3 + 2];
python3.13-pymatting>     
python3.13-pymatting>         int js[4] = {
python3.13-pymatting>             max(    0, x - 1) + y * w,
python3.13-pymatting>             min(w - 1, x + 1) + y * w,
python3.13-pymatting>             x + max(    0, y - 1) * w,
python3.13-pymatting>             x + min(h - 1, y + 1) * w,
python3.13-pymatting>         };
python3.13-pymatting>     
python3.13-pymatting>         float a_sum = 0.0f;
python3.13-pymatting>     
python3.13-pymatting>         for (int d = 0; d < 4; d++){
python3.13-pymatting>             int j = js[d];
python3.13-pymatting>     
python3.13-pymatting>             float da = regularization + fabs(a0 - alpha[j]);
python3.13-pymatting>     
python3.13-pymatting>             a_sum += da;
python3.13-pymatting>     
python3.13-pymatting>             b00 += da * F_prev[j * 3 + 0];
python3.13-pymatting>             b01 += da * F_prev[j * 3 + 1];
python3.13-pymatting>             b02 += da * F_prev[j * 3 + 2];
python3.13-pymatting>     
python3.13-pymatting>             b10 += da * B_prev[j * 3 + 0];
python3.13-pymatting>             b11 += da * B_prev[j * 3 + 1];
python3.13-pymatting>             b12 += da * B_prev[j * 3 + 2];
python3.13-pymatting>         }
python3.13-pymatting>     
python3.13-pymatting>         float a00 = a0 * a0 + a_sum;
python3.13-pymatting>         float a11 = a1 * a1 + a_sum;
python3.13-pymatting>         float a01 = a0 * a1;
python3.13-pymatting>     
python3.13-pymatting>         float inv_det = 1.0f / (a00 * a11 - a01 * a01);
python3.13-pymatting>     
python3.13-pymatting>         F[i * 3 + 0] = fmax(0.0f, fmin(1.0f, inv_det * (a11 * b00 - a01 * b10)));
python3.13-pymatting>         F[i * 3 + 1] = fmax(0.0f, fmin(1.0f, inv_det * (a11 * b01 - a01 * b11)));
python3.13-pymatting>         F[i * 3 + 2] = fmax(0.0f, fmin(1.0f, inv_det * (a11 * b02 - a01 * b12)));
python3.13-pymatting>         B[i * 3 + 0] = fmax(0.0f, fmin(1.0f, inv_det * (a00 * b10 - a01 * b00)));
python3.13-pymatting>         B[i * 3 + 1] = fmax(0.0f, fmin(1.0f, inv_det * (a00 * b11 - a01 * b01)));
python3.13-pymatting>         B[i * 3 + 2] = fmax(0.0f, fmin(1.0f, inv_det * (a00 * b12 - a01 * b02)));
python3.13-pymatting>     }
python3.13-pymatting>     """
python3.13-pymatting>     
python3.13-pymatting> >   platform = cl.get_platforms()[0]
python3.13-pymatting>                ^^^^^^^^^^^^^^^^^^
python3.13-pymatting> E   pyopencl._cl.LogicError: clGetPlatformIDs failed: PLATFORM_NOT_FOUND_KHR
python3.13-pymatting> 
python3.13-pymatting> pymatting/foreground/estimate_foreground_ml_pyopencl.py:97: LogicError
python3.13-pymatting> =========================== short test summary info ============================
python3.13-pymatting> FAILED tests/test_foreground.py::test_foreground - pyopencl._cl.LogicError: clGetPlatformIDs failed: PLATFORM_NOT_FOUND_KHR
python3.13-pymatting> ================== 1 failed, 8 passed, 4 deselected in 40.55s ==================
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.44, NixOS, 25.11 (Xantusia), 25.11.20250903.c6a788f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.29.1`
 - channels(adrian): `""`
 - nixpkgs: `/nix/store/b0vlbbvnq40yca0r0dx39092inxzlp5i-source`



### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/pymatting/default.nix
+++ b/pkgs/development/python-modules/pymatting/default.nix
@@ -52,6 +52,9 @@ buildPythonPackage rec {
     "test_lkm"
   ];
 
+  # pyopencl._cl.LogicError: clGetPlatformIDs failed: PLATFORM_NOT_FOUND_KHR
+  disabledTestPaths = lib.optional cudaSupport "tests/test_foreground.py";
+
   meta = with lib; {
     description = "Python library for alpha matting";
     homepage = "https://github.com/pymatting/pymatting";

